app/routes.py

```diff
@@ -1,2 +1,278 @@
+                and its value is returned in json format
+    """
+    
+    request_body = request.get_json()
+    #checks to see if 'title' is found as a key in request_body called 'validation'
+    if 'title' not in request_body:
+        return make_response(jsonify({"details":"Invalid data"}), 400)
+
+    new_goal = Goal(title = request_body["title"]) 
+    db.session.add(new_goal)
+    db.session.commit()
+        
+    return jsonify({
+    "goal": new_goal.goal_to_python_dict()
+    }), 201
+
+
+@goal_bp.route("/<goal_id>", methods=["GET","PUT","DELETE"],strict_slashes=False)
+def handle_goal(goal_id):
+    """
+        Input: Request to read, edit, or delete a goal by goal_id
+        Output: Returns an error if the goal does not exist, returns dictionary of Goal instance 
+                along with OK 200 code
+    """
+    
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return '',404
+    
+    if request.method == "GET":
+        return jsonify(goal=goal.goal_to_python_dict()), 200
+   
+    if request.method =='PUT':
+        request_body = request.get_json()
+        #assigning updated values to task
+        goal.title = request_body['title']
+        db.session.commit()
+        return jsonify(goal=goal.goal_to_python_dict()), 200
+
+    if request.method == 'DELETE':
+        db.session.delete(goal)
+        db.session.commit()
+        return jsonify(details=f'Goal {goal.goal_id} "{goal.title}" successfully deleted'),200
+
+
+@goal_bp.route('', methods=["GET"],strict_slashes=False)
+def get_goals():
+    """
+        Input: Request to query all saved instances of Goal. 
+        Output: Returns an empty list if there are no saved goals, returns a json dictionary of all 
+                saved goal dictionaries
+    """
+    goals = Goal.query.all()
+    goals_response = []
+    for goal in goals:
+        goals_response.append(goal.goal_to_python_dict())
+    return jsonify(goals_response)
+
+
+
+@goal_bp.route("/<goal_id>/tasks",methods=["POST"], strict_slashes=False)
+def tasks_to_goal(goal_id):
+    """
+        Input:  Request to add a list of task id's to an instance of goal by its goal_id
+        Output: Returns error if goal does not exist, returns a dictionary of the goal instance by its
+                goad_id and the task_ids associated with the goal
+    """
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return '',404
+    # request body has a goal and list of tasks to assigned to that goal
+    request_body = request.get_json()
+    # request_body looks like {'tasks': [1,2]}  value of tasks refers to the task_id columm of Task 
+    for task_id in request_body['task_ids']:
+        task = Task.query.get(task_id)
+        if task not in goal.tasks:
+            goal.tasks.append(task)
+    db.session.commit()
+    
+    return make_response({"id": int(goal_id), "task_ids": request_body["task_ids"]}, 200)                                                                          
```

Comment:
This change would allow the response to include ALL tasks associated with the goal, not just the new ones.
This isn't super intuitive so I'd definitely be down to talk through this more in our next 1:1


```suggestion
     task_list = []
    for task in goal.tasks:
        task_list.append(task.task_id)
        
    return make_response({"id": int(goal_id), "task_ids": task_list}, 200)                                                                          
```